pkg/kubectl/cmd/cp/cp.go

```diff
@@ -477,7 +477,7 @@ func (o *CopyOptions) untarAll(reader io.Reader, destDir, prefix string) error {
 				return err
 			}
 		} else {
-			outFile, err := os.Create(destFileName)
+			outFile, err := os.OpenFile(destFileName, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, mode)
```

Comment:
don't we need to mask the mode to just permission bits by passing in `mode.Perm()`?

Reply:
if we preserve the mode, doesn't it mean we could technically open the file in a non-writable mode? I suspect `cp` does copy the file AND THEN changes the permissions.

Reply:
Also, cp doesn't preserve modes by default AFAIK.

Reply:
`cp` preserve modes in that if you copy an executable it's still executable. Not sure if you mean something else?

Reply:
No, looks like I was wrong :-/